plugins/catalog-backend/src/ingestion/processors/GroupPopulatorProcessor.ts

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Entity, GroupEntity, LocationSpec } from '@backstage/catalog-model';
+import { LocationProcessor, LocationProcessorEmit } from './types';
+import * as result from './results';
+
+export class GroupPopulatorProcessor implements LocationProcessor {
+  async processEntity(
+    entity: Entity,
+    _location: LocationSpec,
+    emit: LocationProcessorEmit,
+  ): Promise<Entity> {
+    if (entity.kind.toLowerCase() !== 'group') {
+      return entity;
+    }
+    const spec = entity.spec as GroupEntity['spec'];
+
+    const { parent, children } = spec;
+    const self = {
+      kind: entity.kind,
+      name: entity.metadata.name,
+      namespace: entity.metadata.namespace,
+    };
+
+    if (parent) {
+      emit(
+        result.relation({
+          type: 'parent',
```

Comment:
hmm such generic names, i was envisioning them to be pretty precise, such as `group-has-parent-group`, `group-has-child-group`, `group-has-user-member`, `user-is-member-of-group`, ... this smells like trouble to me

Reply:
Think I had all kinds of `backstage.io/group-parent` and whatnot, but not what I wanna focus on here